src/primitive.ts

```diff
@@ -126,51 +126,66 @@ export function parseString (str: string, ptr = 0, endPtr = str.length): string
 	return parsed + str.slice(sliceStart, endPtr - 1)
 }
 
-export function parseValue (value: string, toml: string, ptr: number): boolean | number | TomlDate {
+export enum IntegerParsing { NUMBER_OR_ERROR, NUMBER_OR_BIGINT, BIGINT_ONLY }
+
+export function parseValue(value: string, toml: string, ptr: number, integerParsing: IntegerParsing = IntegerParsing.NUMBER_OR_ERROR): boolean | number | bigint | TomlDate {
 	// Constant values
-	if (value === 'true') return true
-	if (value === 'false') return false
-	if (value === '-inf') return -Infinity
-	if (value === 'inf' || value === '+inf') return Infinity
-	if (value === 'nan' || value === '+nan' || value === '-nan') return NaN
+	if (value === 'true') return true;
+	if (value === 'false') return false;
+	if (value === '-inf') return -Infinity;
+	if (value === 'inf' || value === '+inf') return Infinity;
+	if (value === 'nan' || value === '+nan' || value === '-nan') return NaN;
 
-	if (value === '-0') return 0 // Avoid FP representation of -0
+	if (value === '-0') return 0; // Avoid FP representation of -0
```

Comment:
This path must not be taken when we always use bigints.

Reply:
Done, but how about `-0.0` ? That's not handled, is it ?

Reply:
I'm adding a test for that.